### PR TITLE
restrictions to psp

### PR DESCRIPTION
**Pull request by M4itee**
ref #148 
restrictions to the PSP:
- decrease in capabilities,
- privileged setting removed.

### CODE REVIEW

**Review by knatsakis — APPROVED** [review]

**Comment by rsawer** [comment]
did this pass functionality tests?

**Review by ilyam8 — CHANGES_REQUESTED** [review]
Hold on with merging. I need to check it. I doubt that we can remove `privileged` from the sysctlInitContainer. That is the whole purpose of it - to be privileged and to be able to do changes on the host system (not sure tho).

**Comment by M4itee** [comment]
@ilyam8 did you had a time to poke this issue with some kind of stick or not yet? :)

File: charts/netdata/templates/psp.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ metadata:
     chart: {{ .Chart.Name }}-{{ .Chart.Version }}
 spec:
   allowPrivilegeEscalation: true
-  allowedCapabilities:
-  - '*'
+  defaultAddCapabilities:
+    - SYS_PTRACE
```
Comment (ilyam8) [inline charts/netdata/templates/psp.yaml:13]:
```suggestion
  allowedCapabilities:
    - SYS_PTRACE
    - CAP_SYS_ADMIN
```

Those are needed by [apps.plugin](https://github.com/netdata/netdata/tree/master/collectors/apps.plugin#appsplugin) and [cgroups.plugin](https://github.com/netdata/netdata/tree/master/collectors/cgroups.plugin#cgroupsplugin).

See https://github.com/netdata/helmchart/blob/915ea3bd202aef27385638dc73fc61071cecd40b/charts/netdata/templates/daemonset.yaml#L157-L161

And I think that 'allowedCapabilities' is better (more secure) than 'allowedCapabilities'.

File: charts/netdata/templates/psp.yaml [hunk 2]
```diff
@@ -19,7 +19,6 @@ spec:
   hostPorts:
   - max: 65535
     min: 0
-  privileged: true
```
Comment (ilyam8) [inline charts/netdata/templates/psp.yaml:22]: 'privileged: true' is needed for [sysctlInitContainer](https://github.com/netdata/helmchart/blob/915ea3bd202aef27385638dc73fc61071cecd40b/charts/netdata/templates/daemonset.yaml#L48-L56).

I don't know how people use this option (or is it used at all). My understanding is that sysctlInitContainer is supposed to perform changes on the host system thus needs to be priviliged.

**Review by ilyam8 — CHANGES_REQUESTED** [review]

File: charts/netdata/templates/psp.yaml (hunk 2 above)
Comment (ilyam8) [reply charts/netdata/templates/psp.yaml:22]: Related article: https://kubernetes.io/docs/tasks/administer-cluster/sysctl-cluster/

**Review by ShimonOhayon — APPROVED** [review]

**Review by ilyam8 — APPROVED** [review]

**Pull request merged**